### ISSUE TITLE
Remove the "Press ESC to exit" functionality.

### ISSUE THREAD

**Issue by flyklmwl** [kanboard/kanboard#5505]
Hello,

I have a feature suggestion. Every time I write or edit a task, accidentally pressing the ESC button causes an automatic exit. Sometimes I lose a lot of work and have to start over, which is very frustrating. I suggest not associating the ESC button with this action or adding a confirmation prompt before exiting.

Thank you.

**Comment by rautamiekka** [comment]
Personally I'm more for removing instead of a confirmation.

**Comment by colonelpopcorn** [comment]
Wouldn't it be better to just have an on/off switch for keyboard shortcuts?

**Comment by Wc4ever** [comment]
Shortcuts are good, but escape to close the task is crucial for information loose. The most common case how it could happen. You type symbol ":" which type invokes emojis suggesting, you don't want emoji, press escape to close it, task is closed, all description is lost.

**Comment by colonelpopcorn** [comment]
I feel like removing this functionality doesn't make any sense from a backwards compatibility perspective. I would be super mad as a user if I was used to closing modals with the Esc key only to find in an update that it was no longer working. I played around with Tampermonkey and came up with a user script that will remove the Esc key handlers for you. You can either use this user script to remove the functionality for just you, or you should create a plugin that does this for you.

```javascript
// ==UserScript==
// @name         Kanboard turn esc key off
// @namespace    http://tampermonkey.net/
// @version      2024-07-17
// @description  Turn off esc key shortcut for Kanboard
// @author       You
// @match        https://kanboard.yourdomain.net/*
// @icon         https://www.google.com/s2/favicons?sz=64&domain=tampermonkey.net
// @grant        unsafeWindow
// ==/UserScript==

// Replace @match with your domain where kanboard is hosted

// Unsafe window access is required to inject functionality into the window's
// context. Otherwise, KB is not defined and unable to be accessed.
unsafeWindow.Kanboard_removeEscShortcut = setTimeout(
    () => {
        // Reset listeners to a filtered list sans any Escape key events.
        KB.listeners["keys"] = KB.listeners["keys"].filter(x => x.combination !== "Escape");
        return void 0;
    },
    800//ms
);
```

**Comment by cshyam1892** [comment]
I agree that adding a setting to enable or disable the ESC key functionality would be a good idea. Personally, as a Vim user, I find the ESC key a bit intrusive, but I understand that others might prefer to keep it.

For those who want to completely remove the ESC key functionality for modals, you can modify the '**assets/js/components/keyboard-shortcuts.js**' file as follows:

1. Open the **assets/js/components/keyboard-shortcuts.js** file.

2. Search for Escape word in that file and just remove the '**!'** so that it looks like this 
`if (KB.exists('#suggest-menu')) {`

This change will ensure that the ESC key will work for other stuff like dropdown. 


3. Save the file, and execute `./cli js`  in your Kanboard directory to build.

**Comment by raymondmorfe** [comment]
hmmm we can actually create plugin for this, right? just adding the alert() on close button?

**Comment by nekohayo** [comment]
Presuming you are all talking about the overlay-like editing UI that shows up when you click the little edit icon on a card in a board…

…I believe the safe thing to do, which would be consistent with the vast majority of GUIs out there, would be:

* When unsaved changes are present, pressing `Esc` should prompt for confirmation to save or discard the changes.
* If there are no unsaved changes at all, then `Esc` could directly dismiss the overlay dialog.

Then you have the best of both worlds, and no configuration nor plugin needed.

**Comment by YorVeX** [comment]
Came here and found this after losing a lot of text 😢

I was editing something and my text included an @ character. Typing this opened a popup menu to select a username from Kanboard. I didn't want or need this, and in every other application I know, if a hotkey or typing a special key triggers something like this (e.g. intellisense or template options in dev GUIs, or the username selection right here on GitHub when typing an "@"), you press ESC to close it again. So I did that there, but instead of just closing that username selection, the whole editing window with all my unsaved text was gone ☹️

I like the suggestion @nekohayo made the most, that's also a typical workflow I know from other apps and websites. And maybe in addition, when such a popup is open, ESC should refer to this popup and not the editing window (if technically possible).
